src/components/Shared/Filters/NullState/NullState.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { Dispatch, SetStateAction, useState } from 'react';
+import { Box, Typography, Button } from '@material-ui/core';
+import { mdiFormatListBulleted, mdiHome } from '@mdi/js';
+import Icon from '@mdi/react';
+import { Trans, useTranslation } from 'react-i18next';
+import {
+  ContactFilterSetInput,
+  TaskFilterSetInput,
+} from '../../../../../graphql/types.generated';
+import useTaskDrawer from '../../../../../src/hooks/useTaskDrawer';
+import theme from '../../../../theme';
+import { renderDialog } from '../../../Layouts/Primary/TopBar/Items/AddMenu/AddMenu';
+import { NullStateBox } from './NullStateBox';
+
+interface Props {
+  page: 'contact' | 'task';
+  totalCount: number;
+  filtered: boolean;
+  changeFilters:
+    | Dispatch<SetStateAction<TaskFilterSetInput>>
+    | Dispatch<SetStateAction<ContactFilterSetInput>>;
+}
+
+const NullState: React.FC<Props> = ({
+  page,
+  totalCount,
+  filtered,
+  changeFilters,
+}: Props) => {
+  const { t } = useTranslation();
+  const { openTaskDrawer } = useTaskDrawer();
+  const [dialogOpen, changeDialogOpen] = useState(false);
+
+  return (
+    <NullStateBox data-testid={`${page}-null-state`}>
+      <Icon
+        path={page === 'contact' ? mdiHome : mdiFormatListBulleted}
+        size={1.5}
+      />
+      {filtered ? (
+        <>
+          <Typography variant="h5">
+            <Trans
+              defaults="You have {{count}} total {{page}}s"
```

Comment:
This isn't a very robust way to translate as plurals probably don't work this way in every language... Maybe a simple way would be to pass in a plural prop too.

Reply:
I was reading the doc for react-i18n and it seems for plurals they normally use the _plural extension on the key in the translation json, but since one of the plurals is a variable itself, it doesn't seem to work and they don't provide an example on this case so I'm not sure if its possible. The way I just changed it to is passing a page or page+s when needed which should be fine since its either contact/contacts or task/tasks. One way to get the key thing work would be to have separate keys for contacts and tasks so that we don't need the page variable.

Reply:
I think your new solution isn't as good as your old version, although it didn't support the singular case. Before a translator at least had control over the string. Now it's part of the app's logic. I think I'm ok with merging your previous version and fixing it later.

Ideally I think it would be best to translate the word as a whole instead of tacking on characters. I guess you'd have to do that where you pass in the `page` prop. Idk how to do that with plurals other than passing in 2 props.

Guess we still have some translation stuff to figure out...

Reply:
Alright sure, by passing another prop you mean like passing in both "contact" and "contacts"?

Reply:
Ya I guess so. That doesn't seem very clean... 🤷‍♂️